### PR TITLE
Util carto shapefile

### PR DESCRIPTION
**Pull request by adawyj97**

### CODE REVIEW

File: utils/util_carto.py [hunk 1]
```diff
@@ -17,8 +20,96 @@ def upload_to_carto(file, privacy):
...
-    # set dataset privacy to 'Public with link'
+    # set dataset privacy
+    dataset.privacy = privacy
+    dataset.save()
+
+def create_carto_schema(gdf):
+    '''
+    Function to create a dictionary of column names and data types
+    in order to the upload the data to Carto
+    INPUT   gdf: geodataframe storing the data (dataframe)
```
Comment (amsnyder) [inline utils/util_carto.py:31]: Does this have to be a geodataframe or could it be a regular dataframe?
```suggestion
    INPUT   gdf: geodataframe storing the data (dataframe)
```

File: utils/util_carto.py [hunk 2]
```diff
@@ -17,8 +20,96 @@ def upload_to_carto(file, privacy):
...
+    RETURN  ouput: an ordered dictionary (dictionary of strings)
+    '''
+    # create an empty list to store column names
+    list_cols = []
+    # column names and types for data table
```
Comment (amsnyder) [inline utils/util_carto.py:36]:
```suggestion
    # get column names and types for data table
```

File: utils/util_carto.py (hunk 1 above)
Comment (adawyj97) [reply utils/util_carto.py:31]:
```suggestion
    INPUT   gdf: dataframe storing the data (dataframe)
```

File: utils/util_carto.py [hunk 3]
```diff
@@ -17,8 +20,96 @@ def upload_to_carto(file, privacy):
...
+    # column types should be one of the following: geometry, text, numeric, timestamp
+    for col in list(gdf):
```
Comment (amsnyder) [inline utils/util_carto.py:38]:
```suggestion
    for col in list(df):
```

File: utils/util_carto.py [hunk 4]
```diff
@@ -17,8 +20,96 @@ def upload_to_carto(file, privacy):
...
+        # if the column type is float64 or int64, assign the column type as numeric
+        if (gdf[col].dtypes  == 'float')| (gdf[col].dtypes  == 'int'):
```
Comment (amsnyder) [inline utils/util_carto.py:40]:
```suggestion
        if (df[col].dtypes  == 'float')| (df[col].dtypes  == 'int'):
```

**Pull request merged**